[MM-32576] Text copied from Microsoft OneNote pastes as an image (v2)

akaravashkin opened the pull request:
#### Summary
Added ability to paste plain text while copy-pasting from MS Office products on Mac, **WITHOUT** picture.

This is a second version if this PR: https://github.com/mattermost/mattermost-server/pull/23401
[hmhealey](https://github.com/hmhealey) suggested to drop an image and save only text in this case.

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/16858
https://mattermost.atlassian.net/browse/MM-32576

#### Screenshots


#### Release Note
```release-note
NONE
```

mattermost-build commented:
Hello @akaravashkin,

Thanks for your pull request! A Core Committer will review your pull request soon. For code contributions, you can learn more about the review process [here](https://developers.mattermost.com/contribute/getting-started/code-review/).

mattermost-build commented:
E2E tests not automatically triggered, because PR has no approval yet. Please ask a developer to review and then try again to attach the QA label.

akaravashkin commented:
> Would you mind fixing that linting issue? It just wants you to remove the now-unused import
Thank you for noting this. I made another commit. Also, I found a problem with tests for this file, I fixed them too.

M-ZubairAhmed commented:
/update-branch

mattermost-build commented:
We don't have permissions to update this PR, please contact the submitter to apply the update.

yasserfaraazkhan commented:
/e2e-test

mattermost-build commented:
Successfully triggered E2E testing!
[GitLab pipeline](https://git.internal.mattermost.com/qa/cypress-ui-automation/-/pipelines/344568) | [Test dashboard](https://automation-dashboard.vercel.app/cycle/344568)

akaravashkin commented:
@M-ZubairAhmed I rebased my branch with latest master and pushed with force

M-ZubairAhmed commented:
/update-branch

mattermost-build commented:
We don't have permissions to update this PR, please contact the submitter to apply the update.

yasserfaraazkhan commented:
/e2e-test

mattermost-build commented:
Successfully triggered E2E testing!
[GitLab pipeline](https://git.internal.mattermost.com/qa/cypress-ui-automation/-/pipelines/345508) | [Test dashboard](https://automation-dashboard.vercel.app/cycle/345508)

yasserfaraazkhan reviewed (changes requested):
Hi @akaravashkin , @M-ZubairAhmed 
we were seeing some errors while executing test under `tests/integration/channels/files_and_attachments/*.js`, can you check if that needs a fix?

<img width="1287" alt="Screenshot 2023-06-16 at 9 55 25 PM" src="https://github.com/mattermost/mattermost/assets/15979783/f54e459f-ca03-4d0e-a1a4-7d67c41c8944">

<img width="928" alt="Screenshot 2023-06-16 at 9 58 06 PM" src="https://github.com/mattermost/mattermost/assets/15979783/fbd0b2c7-1085-4813-9bfc-b8aff4f11978">

M-ZubairAhmed commented:
> Hi @akaravashkin , @M-ZubairAhmed we were seeing some type errors like `clipboardData.getData is not a function` while executing test under `tests/integration/channels/files_and_attachments/*.js`, can you check if that needs a fix?
> 
> <img alt="Screenshot 2023-06-16 at 9 55 25 PM" width="1287" src="https://user-images.githubusercontent.com/15979783/246474604-f54e459f-ca03-4d0e-a1a4-7d67c41c8944.png"> <img alt="Screenshot 2023-06-16 at 9 58 06 PM" width="928" src="https://user-images.githubusercontent.com/15979783/246475204-fbd0b2c7-1085-4813-9bfc-b8aff4f11978.png">

Hi @akaravashkin can you please fix the broken e2es

akaravashkin commented:
@M-ZubairAhmed @yasserfaraazkhan I mocked clipboardData.getData() function for e2e tests and updated my branch

yasserfaraazkhan commented:
/e2e-test

mattermost-build commented:
Successfully triggered E2E testing!
[GitLab pipeline](https://git.internal.mattermost.com/qa/cypress-ui-automation/-/pipelines/346187) | [Test dashboard](https://automation-dashboard.vercel.app/cycle/346187)

mattermost-build commented:
This PR has been automatically labelled "stale" because it hasn't had recent activity.
A core team member will check in on the status of the PR to help with questions.
Thank you for your contribution!

hmhealey commented:
@akaravashkin Our CI pipeline seems to be having some issues, and we unfortunately didn't get back to this before a conflicting PR went in.

Would you mind updating this to the latest master again? If you could also rename your fork from `mattermost-server` to just `mattermost`, that might help as well. We've seen occasional issues with PRs coming from forks since we renamed this repo a while back

akaravashkin commented:
@hmhealey I updated my branch with latest master and also renamed the fork. It was a conflict and I decided to keep my version. We should skip paste not only if html table presented, but if got any `text/html` data

<img width="788" alt="image" src="https://github.com/mattermost/mattermost/assets/1384262/39ff72f8-8beb-4d38-b8ba-2d4309ebb7b2">

M-ZubairAhmed commented:
/update-branch

mattermost-build commented:
We don't have permissions to update this PR, please contact the submitter to apply the update.

M-ZubairAhmed commented:
/update-branch

mattermost-build commented:
We don't have permissions to update this PR, please contact the submitter to apply the update.

M-ZubairAhmed reviewed (changes requested):
Sorry taking a look at this again. It looks like the correct place to add this functionality is not in `file_upload` but in the paste handler of advance_create_comment and advance_create_post. Let me know if you are up for doing this if not I can take it over

yasserfaraazkhan commented:
/e2e-test

mattermost-build commented:
Successfully triggered E2E testing!
[GitLab pipeline](https://git.internal.mattermost.com/qa/cypress-ui-automation/-/pipelines/351118) | [Test dashboard](https://automation-dashboard.vercel.app/cycle/351118)

hmhealey commented:
> @hmhealey I updated my branch with latest master and also renamed the fork. It was a conflict and I decided to keep my version. We should skip paste not only if html table presented, but if got any `text/html` data
> <img alt="image" width="788" src="https://user-images.githubusercontent.com/1384262/250138030-39ff72f8-8beb-4d38-b8ba-2d4309ebb7b2.png">

Yeah, that was the right thing to do. The only reason that conflicted was because the `getTable` function was renamed to `getHtmlTable`, so it's safe to use your code.

> Sorry taking a look at this again. It looks like the correct place to add this functionality is not in `file_upload` but in the paste handler of advance_create_comment and advance_create_post. Let me know if you are up for doing this if not I can take it over

@M-ZubairAhmed What's here is actually correct. The reason this isn't working on master is because the `FileUpload` component intercepts the paste event before it can get to the post textbox, and this change makes it so that the `FileUpload` doesn't interfere with someone pasting anything containing raw HTML. In other words, this is a bug in the `FileUpload` component rather than an improvement to `AdvancedCreatePost`/`AdvancedCreateComment`

M-ZubairAhmed commented:
/update-branch

mattermost-build commented:
We don't have permissions to update this PR, please contact the submitter to apply the update.

marianunez commented:
Doing some internal reviews - Adding `Do Not Merge` for now.

mattermost-build commented:
This PR has been automatically labelled "stale" because it hasn't had recent activity.
A core team member will check in on the status of the PR to help with questions.
Thank you for your contribution!

M-ZubairAhmed commented:
/update-branch

mattermost-build commented:
We don't have permissions to update this PR, please contact the submitter to apply the update.

M-ZubairAhmed commented:
/update-branch

mattermost-build commented:
We don't have permissions to update this PR, please contact the submitter to apply the update.

marianunez commented:
@akaravashkin we are good to merge this now. Thanks again for your contribution and patience!

amyblais commented:
/cherry-pick cloud

amyblais commented:
/cherry-pick release-8.1

mattermost-build commented:
Cherry pick is scheduled.
